Models/PFUser+ExtendedUser.m

```diff
@@ -124,6 +131,13 @@ - (void)retrieveFavoritesWithCompletion:(void(^)(NSArray<GMSPlace*>*))completion
 
 - (void)retrieveWishlistWithCompletion:(void(^)(NSArray<GMSPlace*>*))completion {
     NSMutableArray<GMSPlace*>* places = [NSMutableArray new];
+   
+    // if there's no wishlist, call completion with nothing
+    if(self.wishlist.count <= 0)
+    {
+        completion(nil);
```

Comment:
Same question with passing nil here. I'm asking because in general, the less special cases (e.g. empty wishlist or favorites count) we have, the simpler our code becomes

Reply:
I set it to pass nil so that when we use the block somewhere else, if it's nil then we know there are no favorites/wishlist to retrieve, and inside the completion block we can act differently depending on that. For example, if it's not nil then we'll show the tableview, if it's nil we'll show the "No favorites" message.

Should we pass in an empty array, and in the block definition, instead of checking if it's nil, check if it's an empty array? Or maybe it's better to remove this check from PFUser and simply do something like:
if(self.favorites.count > 0)
    call the 'retrieveFavorites...' method
else
    hide tableview and show "No favorites" message

outside of the PFUser class when trying to retrieve followers/wishlist.

Reply:
I think the latter sounds best, but let's chat about it tomorrow so I can get full context!

Reply:
okay :D thanks